### PR TITLE
chore: Log In

### PR DESCRIPTION
**Pull request by SauveJeanLuc**
# Add login functionality for driver and operator roles

## Description

This pull request implements the login feature for all roles. It adds a login form with validation and error handling, authentication logic using JWT and bcrypt.

## How to test

- Run `npm install` to install the dependencies
- Run `npm run dev` to start the development server
- Navigate to `http://localhost:{PORT}/api-docs` for swagger documentation and head to Auth on auth/login endpoint.
-  Enter the credentials provided by the operator or administrator
- Verify that you can log in successfully and access the tasks assigned to your role
- Verify that you cannot log in with invalid credentials or access tasks that are not authorized for your role

## Screenshots

image
image

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added comments and documentation where necessary
- [x] I have followed the coding standards and style guidelines

### CODE REVIEW

**Review by thepatrickniyo — COMMENTED** [review]
LGTM though tests are still missing for this work.

File: src/db/models/user.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      email: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true,
+      },
+      password: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      role: {
```
Comment (TuyizeeAnastase) [inline src/db/models/user.js]: the role is integer, not string name it as `role_id:integer`

File: src/api/validations/auth.validator.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as Joi from "joi";
+import { StatusCodes } from "http-status-codes";
+
+export const validateLogin = (req, res, next) => {
+  const schema = Joi.object({
+    email: Joi.string().email().required(),
+    password: Joi.string().required(),
```
Comment (TuyizeeAnastase) [inline src/api/validations/auth.validator.js]: validate password length check `max` function

File: src/api/controllers/auth.controller.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { StatusCodes } from "http-status-codes";
+import { verifyPassword } from "../utils/hash-password";
+import { signJwtToken } from "../utils/jwt";
+
+import models from "../../db/models";
+export const login = async (req, res) => {
+
+    const { email, password } = req.body;
+  
+    const user = await models.user.findOne({ where: { email } }); // find the user by email
```
Comment (TuyizeeAnastase) [inline src/api/controllers/auth.controller.js]: create services folder where we can add all of functions to access database

File: src/api/controllers/auth.controller.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+    if (!user)
+      return res
+        .status(StatusCodes.UNAUTHORIZED)
+        .json({ success: false, message: "Invalid credentials" });
+    const isPassValid = await verifyPassword(password, user.password);
+
```
Comment (TuyizeeAnastase) [inline src/api/controllers/auth.controller.js]: add middleware to check if user exist

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]

File: src/db/models/user.js (hunk 1 above)
Comment (SauveJeanLuc) [reply src/db/models/user.js]: Done.

File: src/api/validations/auth.validator.js (hunk 2 above)
Comment (SauveJeanLuc) [reply src/api/validations/auth.validator.js]: Done.

File: src/api/controllers/auth.controller.js (hunk 3 above)
Comment (SauveJeanLuc) [reply src/api/controllers/auth.controller.js]: Done.

File: src/api/controllers/auth.controller.js (hunk 4 above)
Comment (SauveJeanLuc) [reply src/api/controllers/auth.controller.js]: Done.

**Review by thepatrickniyo — COMMENTED** [review]
I can see your tests are failing on github due to failure to recognize DB environments. Can you please check in my branch `/github/workflows/ci-cd.yml` There is one line you need to add to get that fixed.

**Review by ernesteNtezirizaza — APPROVED** [review]

**Review by thepatrickniyo — APPROVED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Pull request merged**